src/fungible_token.rs

```diff
@@ -31,43 +42,98 @@ impl FungibleToken {
     pub fn new() -> Self {
         Self {
             total_supply: 0,
+            total_supply_near: 0,
+            total_supply_eth: 0,
             account_storage_usage: 0,
         }
     }
 
+    /// Balance of NEAR tokens
     pub fn internal_unwrap_balance_of(&self, account_id: AccountId) -> Balance {
         match self.accounts_get(account_id) {
             Some(balance) => u128::try_from_slice(&balance[..]).unwrap(),
-            None => sdk::panic_utf8("The account is not registered".as_bytes()),
+            None => sdk::panic_utf8(b"ERR_ACCOUNT_NOT_EXIST"),
         }
     }
 
+    /// Balance of ETH tokens
+    pub fn internal_unwrap_balance_of_eth(&self, address: EthAddress) -> Balance {
+        Engine::get_balance(&prelude::Address(address)).as_u128()
+    }
+
+    /// Internal deposit NEAR (nETH) FT
     pub fn internal_deposit(&mut self, account_id: AccountId, amount: Balance) {
         let balance = self.internal_unwrap_balance_of(account_id.clone());
         if let Some(new_balance) = balance.checked_add(amount) {
             self.accounts_insert(account_id, new_balance);
+            self.total_supply_near = self
+                .total_supply_near
+                .checked_add(amount)
+                .expect("ERR_TOTAL_SUPPLY_OVERFLOW");
             self.total_supply = self
                 .total_supply
                 .checked_add(amount)
-                .expect("Total supply overflow");
+                .expect("ERR_TOTAL_SUPPLY_OVERFLOW");
         } else {
-            sdk::panic_utf8("Balance overflow".as_bytes());
+            sdk::panic_utf8(b"ERR_BALANCE_OVERFLOW");
         }
     }
 
+    /// Internal deposit ETH FT
+    pub fn internal_deposit_eth(&mut self, address: EthAddress, amount: Balance) {
+        let balance = self.internal_unwrap_balance_of_eth(address.clone());
+        if let Some(new_balance) = balance.checked_add(amount) {
+            Engine::set_balance(&prelude::Address(address), &U256::from(new_balance));
+            self.total_supply_eth = self
+                .total_supply_eth
+                .checked_add(amount)
+                .expect("ERR_TOTAL_SUPPLY_OVERFLOW");
```

Comment:
Question: is there a difference between `expect` and `sdk::panic_utf8`?

If not, then it seems like it would be nice to be consistent in which we use. E.g.

```rust
let new_balance = balance.checked_add(amount).expect("ERR_BALANCE_OVERFLOW");
// other check_add calls ...
```

If there is a difference then I am interested in knowing what the difference is and when it is appropriate to use either.